SmartDeviceLink/private/SDLIAPDataSession.m

```diff
@@ -125,6 +144,34 @@ - (BOOL)sdl_isIOThreadCancelled {
     return NO;
 }
 
+- (void)sdl_retryThreadCancel {
+    if (self.ioStreamThread == nil) {
+        SDLLogV(@"No data session established during Retry");
+        [super cleanupClosedSession];
+    }
+    SDLLogE(@"Will retry thread cancel in %.1f seconds", IOStreamThreadRetryWaitSecs);
+    __weak typeof(self) weakSelf = self;
+    dispatch_async(dispatch_get_main_queue(), ^{
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+
+        // Attempt to cancel the ioStreamThread. Once the thread realizes it has been cancelled, it will cleanup the I/O streams. Make sure to wake up the run loop in case there is no current I/O event running on the ioThread.
+        [strongSelf.ioStreamThread cancel];
+        [strongSelf performSelector:@selector(sdl_doNothing) onThread:self.ioStreamThread withObject:nil waitUntilDone:NO];
+
+        // Block the thread until the semaphore has been released by the ioStreamThread (or a timeout has occured).
+        BOOL cancelledSuccessfully = [strongSelf sdl_isIOThreadCancelled];
+        if (!cancelledSuccessfully) {
+            SDLLogE(@"The I/O streams were not shut down successfully.  Will try again.");
+            [strongSelf performSelector:@selector(sdl_retryThreadCancel) withObject:nil afterDelay:IOStreamThreadRetryWaitSecs];
+            [strongSelf sdl_retryThreadCancel];
+        } else {
+            strongSelf.ioStreamThread = nil;
+        }
+
+        [strongSelf.sendDataQueue removeAllObjects];
+    });
+}
+
```

Comment:
As already mentioned for `destroySessionWithCompletionHandler` this method is not necessary. It's a copy and instead the destroy method should be called.
```suggestion
```